### ISSUE TITLE
`concave_hull` crashes with specific `MultiPoint` input

### ISSUE THREAD

**Issue by JuriaanSioux** [shapely/shapely#1971]
## Expected behavior and actual behavior.

The underlying concave hull code crashes the python interpreter completely. I understand some of the points are very close together, which might make it a problematic case, but I would at the very least expect an exception rather than a complete interpreter crash.

Looking at the implementation, I would think the problem occurs somewhere in the c bindings or GEOS itself, but I know too little about this to say anything constructive.

## Steps to reproduce the problem.

```python
import shapely

points = [
    [175682.210293, 505233.365178],
    [175682.210295, 505233.365177],
    [175684.375108, 505254.524644],
    [175684.375109, 505254.524642],
    [175683.226468, 505243.284373],
]
shapely.concave_hull(
    shapely.MultiPoint(points)
)  # this crashes with "Process finished with exit code -1073741819 (0xC0000005)"

```

## Operating system

Windows 10

## Shapely version and provenance

2.02 installed from PyPI using pip with `shapely.geos_version` returning `(3, 11, 2)`.

**Comment by jorisvandenbossche** [comment]
Testing locally where I had GEOS 3.12.0, I could reproduce the crash, but after updating to 3.12.1, it's working. So it seems to be fixed only in the latest bug-fix release.

I am wondering if this would also be fixed in 3.11.3 (which was released along 3.12.1). I can't quickly test because it is not available on conda-forge. But if we do another 2.0.x release, we can update GEOS to 3.11.3

**Comment by mwtoews** [comment]
Testing this, I can confirm that it was fixed for GEOS 3.12.1. But it was not fixed for GEOS 3.11.3. There are no obvious [release notes for 3.12](https://github.com/libgeos/geos/blob/3.12/NEWS.md) to show what fixed the issue, so it's not easy to track down and backport to earlier branches.
Tracked upstream with https://github.com/libgeos/geos/issues/1034

**Comment by mwtoews** [comment]
This is a duplicate of #1873 with upstream fixes on the way.

**Comment by mwtoews** [comment]
Fixed with GEOS 3.11.4 (when released)

**Comment by mwtoews** [comment]
Fixed confirmed with shapely-2.0.5
